mantidproject/mantid
# ConvFit file input validation

Fixes #14290

The function that extends the workspace resolution now runs when the algorithm runs not when data is loaded in. This helps to prevent some of the previous issues (see #14290)

**Files are available from `\\olympic\babylon5\Public\ElliotOram\Water`**
# To Test
- Turn on the option in Mantid to show hidden workspaces (View > Preferences... > Mantid > Options > Check `Show Invisible Workspaces` option)
- Open ConvFit (Interfaces > Indirect > Data Analysis > ConvFit)
- Sample = `irs26176_graphite002_red.nxs`
- Resolution = `irs26173_graphite002_res.nxs`
- Ensure that only the `Load` algorithms take place. _(previously this would have run append workspace)_
- Change Fit Type to `One Lorentzian` in the drop down menu
- Click `Run` and wait for the algorithm to conclude
  - Ensure that it produces no errors 
- Delete the following workspaces from the ADS:
  - `__convfit_fit_ws`
  - `__ConvFit_Resolution`
  - `__TMP000000...` _(this file is being removed in #14625 )_
  - `irs26173_graphite002_res`
- Edit the file path of the Resolution file:
  - It should currently be `\\olympic\babylon5\Public\ElliotOram\Water\irs26173_graphite002_res.nxs`
  - Change it to: `\\olympic\babylon5\Public\ElliotOram\Water\irs26174_graphite_red.nxs` **Do NOT press enter, click off the field, ect. This is to ensure the file is NOT loaded into the ADS straight away**
- Ensure `irs26174_graphite002_red` is **NOT** loaded into the ADS
  - if it is delete it and repeat the step from _Edit the file path of the Resolution file:_
- Click `Run`
- Ensure that the algorithm runs without error
